greenplumpython/table.py

```diff
@@ -503,7 +500,61 @@ def _build_full_query(self) -> str:
             return self._query
         return "WITH " + ",".join(cte_list) + self._query
 
-    def fetch(self, is_all: bool = True) -> Iterable[Tuple[Any]]:
+    def __iter__(self) -> "Table":
+        if self._contents is not None:
+            self._n = 0
+            return self
+        assert self._db is not None
+        result = self._fetch()
+        assert result is not None
+        self._contents: List[RealDictRow] = list(result)
+        self._n = 0
+        return self
+
+    def __next__(self):
+        def detect_duplicate_keys(json_pairs: List[tuple[str, Any]]):
+            key_count = collections.Counter(k for k, _ in json_pairs)
+            duplicate_keys = ", ".join(k for k, v in key_count.items() if v > 1)
+
+            if len(duplicate_keys) > 0:
+                raise Exception("Duplicate column_name(s) found: {}".format(duplicate_keys))
+
+        def validate_data(json_pairs: List[tuple[str, Any]]):
+            detect_duplicate_keys(json_pairs)
+            return dict(json_pairs)
+
+        if self._n < len(self._contents):
+            row_contents: Dict[str, Union[str, List[str]]] = {}
+            assert self._contents is not None
+            for name in self._contents[0].keys():
+                if name == "to_json":
+                    to_json_dict = json.loads(
+                        self._contents[self._n][name], object_pairs_hook=validate_data
+                    )
+                    for sub_name in to_json_dict:
+                        row_contents[sub_name] = to_json_dict[sub_name]
+                else:
+                    row_contents[name] = self._contents[self._n][name]
+            self._n += 1
+            return Row(row_contents)
+        raise StopIteration("StopIteration: Reached last row of table!")
+
+    def refresh(self) -> "Table":
+        """
+        Refresh self._contents
+
+        Returns:
+            self
+        """
+
+        assert self._db is not None
+        result = self._fetch()
+        assert result is not None
+        self._contents = list(result)
+        self._n = 0
+        return self
```

Comment:
seems not test for this func and maybe we make it private like `_refresh` better?

Reply:
I did add but it is disappeared, let me add it again